packages/x-data-grid/src/hooks/features/rows/useGridRowSpanning.ts

```diff
@@ -179,7 +178,8 @@ export const rowSpanningStateInitializer: GridStateInitializer = (state, props,
       !orderedFields.length ||
       !dataRowIdToModelLookup ||
       !columnsLookup ||
-      isFilteringPending
+      isFilteringPending ||
+      !props.rowSpanning
```

Comment:
The `if` condition above handles it already, right?
```suggestion
      isFilteringPending
```

Reply:
Right, didn't see it for some reason. Changed it to early return instead for better readability.